src/highlighting.c

```diff
@@ -1672,6 +1749,10 @@ gboolean highlighting_is_comment_style(gint lexer, gint style)
 				style == SCE_C_PREPROCESSORCOMMENT ||
 				style == SCE_C_PREPROCESSORCOMMENTDOC ||
 				style == SCE_C_COMMENTLINEDOC ||
+				/* FIXME: this is sub-stylable, but we can't figure out the base
+				 * style without the Scintilla instance -- and we can't simply
+				 * assume that if the style ID is larger than the max possible
+				 * one it's this, because SCE_C_IDENTIFIER is also sub-stylable... */
```

Comment:
Should it actually be per Scintilla instance, the plugin might only mess with one?

Reply:
It is, so the yeah, we'd have to be sure allocation is the same for all lexer instances.  That's the tricky part, then we can just fill something when initially allocating the substyles.

The proper solution should be `style = SCI_GETSTYLEFROMSUBSTYLE(style)` but we can't do that with this API.

Reply:
Also, don't we only instantiate one lexer per sci instance?  So the whole process needs to be done every time a new tab with a new filetype is opened because there is now a new type of lexer to query?

Reply:
Yes we do have one lever per instance, that's why we'd need the allocations to be the same for all of them for this to have a chance of working.

But really, I probably should look at callers for this to see if anybody actually doesn't have a Scintilla instance ready at hand when calling this.

Reply:
Yeah, it should only be called on a document, so there should be a SCI available.

Reply:
Fixed in e9ca1025d by introducing an alternate API and deprecating this